src/consensus/blockstore.rs

```diff
@@ -203,51 +255,53 @@ impl Blockstore for BlockstoreImpl {
             .map(|c| c.0)
     }
 
-    /// Gives the number of stored shreds for a given `slot` (across all slices).
-    fn stored_shreds_for_slot(&self, slot: Slot) -> usize {
-        self.slot_data(slot)
-            .map_or(0, |s| s.canonical.shreds.values().map(Vec::len).sum())
+    /// Gives reference to stored block for the given `block_id`.
+    ///
+    /// Returns `None` if blockstore does not hold that block.
+    fn get_block(&self, block_id: BlockId) -> Option<&Block> {
+        let block_data = self.get_block_data(block_id)?;
+        block_data.completed.as_ref().map(|(_, block)| block)
```

Comment:
nit: should we add a debug assert to check that block_id.hash matches the hash on the block_data?  It will help the reader understand the properties of how the data is stored mainly.